core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -1107,8 +1107,10 @@ private[kafka] class Processor(
     val header = RequestHeader.parse(buffer)
     if (apiVersionManager.isApiEnabled(header.apiKey, header.apiVersion)) {
       header
-    } else {
+    } else if (header.isApiVersionDeprecated()) {
```

Comment:
Should we add the deprecated versions back to the JSON files? This would enable us to provide more informative error messages to clients, helping them understand why certain APIs are marked as "unsupported" (deprecated, removed, or disabled).

Reply:
No, I added a commit at the end that changes this incorrectly:

https://github.com/apache/kafka/pull/18295/commits/b381c07a595ad616aca2579fb34660f48ea60333

Thanks for catching that.

Reply:
I don't think we need that, see the response to your other comment. Does that make sense?

Reply:
> Does that make sense?

yes, that makes sense :)